File: queue-api/src/main/java/net/mythicisland/queue/api/match/Assignment.java
package net.mythicisland.queue.api.match;

/**
 * The game server a match was allocated to.
 *
 * @param serverId the unique ID of the server
 * @param serverName the name used to connect players, for example battle-1
 */
public record Assignment(
        String serverId,
        String serverName
) {
}

File: queue-shared/src/main/kotlin/net/mythicisland/queue/shared/match/Assignment.kt
package net.mythicisland.queue.shared.match

import build.buf.gen.mythicisland.queue.v2.assignment

data class Assignment(
    val serverId: String,
    val serverName: String,
) {

    fun toDefinition(): build.buf.gen.mythicisland.queue.v2.Assignment {
        return assignment {
            serverId = this@Assignment.serverId
            serverName = this@Assignment.serverName
        }
    }

}
